### PR TITLE
feat(clerk-js): Retheme design and arrow on primary button

### PR DESCRIPTION
**Pull request by marcelscruz**
## Description

This PR adds the new design and arrow on the primary button.

Screenshot 2023-11-13 at 15 00 12

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/clerk-js/src/ui.retheme/icons/arrow-right.svg
Comment (anagstef) [inline packages/clerk-js/src/ui.retheme/icons/arrow-right.svg:1]: This icon is also used in `packages/clerk-js/src/ui/components/OrganizationList/shared.tsx`. Do we want to change it there?

Comment (marcelscruz) [reply packages/clerk-js/src/ui.retheme/icons/arrow-right.svg:1]: Good catch.
I'll create a separate file for the moment to avoid any issues.

Comment (marcelscruz) [reply packages/clerk-js/src/ui.retheme/icons/arrow-right.svg:1]: Done.
Named it `arrow-right-button.svg` for now.
Let's revisit it later and improve on this name or even replace the old arrow once we have more context about the re-design.

**Review by tmilewski — COMMENTED** [review]
@marcelscruz 🙃 Am I crazy or does the arrow look off-center (vertically) from the text?

**Comment by marcelscruz** [comment]
> @marcelscruz 🙃 Am I crazy or does the arrow look off-center (vertically) from the text?

It does indeed, but it's the same as the Figma file.
Maybe we should address that? cc @PixelJanitor

Screenshot 2023-11-15 at 15 39 10

**Review by panteliselef — APPROVED** [review]

File: packages/clerk-js/src/ui.retheme/primitives/Button.tsx [hunk 1]
```diff
@@ -130,7 +143,7 @@ const { applyVariants, filterProps } = createVariants((theme, props: OwnProps) =
       textVariant: 'buttonRegularRegular',
       colorScheme: 'primary',
       variant: 'solid',
-      size: 'md',
+      size: 'sm',
```
[the remark this answers is not in the document]
Comment (marcelscruz) [reply packages/clerk-js/src/ui.retheme/primitives/Button.tsx:146]: Based on the current designs we have, it seems all buttons are smaller indeed.
we could instead change the values of `md`, but they'd end up being the same as `sm`.
Until we have a defined styled system and all component's designs it's hard for us to decide how to reuse our current system.
Wdyt?

File: packages/clerk-js/src/ui.retheme/primitives/Button.tsx [hunk 2]
```diff
@@ -193,7 +207,25 @@ const Button = React.forwardRef<HTMLButtonElement, ButtonProps>((props, ref) =>
...
+          gap={2}
+        >
+          {children}
+          {hasArrow && (
+            <Icon
+              icon={ArrowRightButtonIcon}
+              sx={{
+                width: '0.625rem',
+                height: '0.625rem',
+                opacity: 0.6,
```
[the remark this answers is not in the document]
Comment (marcelscruz) [reply packages/clerk-js/src/ui.retheme/primitives/Button.tsx]: Good catch, thanks!

File: packages/clerk-js/src/ui.retheme/primitives/Button.tsx [hunk 3]
```diff
@@ -69,9 +73,18 @@ const { applyVariants, filterProps } = createVariants((theme, props: OwnProps) =
         solid: {
           backgroundColor: vars.accent,
           color: theme.colors.$colorTextOnPrimaryBackground,
-          '&:hover': { backgroundColor: vars.accentDark },
+          boxShadow: `0px 0px 0px 1px ${theme.colors.$primary800}, 0px 1px 1px 0px rgba(255, 255, 255, 0.07) inset, 0px 2px 3px 0px rgba(34, 42, 53, 0.20), 0px 1px 1px 0px rgba(0, 0, 0, 0.24)`,
```
[the remark this answers is not in the document]
Comment (marcelscruz) [reply packages/clerk-js/src/ui.retheme/primitives/Button.tsx]: I've moved to a [separate util inside `common.ts`](https://github.com/clerk/javascript/pull/2121/files#diff-bd26bc7e587f98ccc6a2e9ce942628855b9e6ed7b0b587c1c6f47fb29fa98c1dR180-R183), as it needs to parse the dynamic value based on the current theme.
Let me know if you see a better solution for it.

File: packages/clerk-js/src/ui.retheme/primitives/Button.tsx [hunk 4]
```diff
@@ -69,9 +73,18 @@ const { applyVariants, filterProps } = createVariants((theme, props: OwnProps) =
...
           '&:focus': props.hoverAsFocus ? { backgroundColor: vars.accentDark } : undefined,
-          '&:active': { backgroundColor: vars.accentDarker },
+          // '&:active': { backgroundColor: vars.accentDarker }, // TODO: once we have the new color palette
+          ':after': {
+            position: 'absolute',
+            content: '""',
+            borderRadius: 'inherit',
+            zIndex: -1,
+            inset: 0,
+            background: 'linear-gradient(180deg, rgba(255, 255, 255, 0.10) 0%, rgba(255, 255, 255, 0.00) 100%)',
```
Comment (anagstef) [inline packages/clerk-js/src/ui.retheme/primitives/Button.tsx:86]: Can we also move this literal values?

Comment (desiprisg) [reply packages/clerk-js/src/ui.retheme/primitives/Button.tsx:86]: We need to use `colorPrimary` somehow here, in order for this to be themable.

Comment (desiprisg) [reply packages/clerk-js/src/ui.retheme/primitives/Button.tsx:86]: Nevermind, didn't understand that correctly. This property actually creates the shadow effect on the button, so no need to change anything

**Pull request merged**